src/format.ts

```diff
@@ -110,8 +126,137 @@ function checkFormat(srcFiles: string[], baseArgs: string[]): Promise<boolean> {
     });
   });
 }
+
+/**
+ * Parses through xml string for the replacement string, offset, length and
+ * returns all of the necessary replacements for the file
+ *
+ * @param output xml string from clangFormat
+ */
+function getReplacements(fileXML: string): Replacement[] {
+  const replacements: Replacement[] = [];
+
+  let xmlLines = fileXML.split('\n');
+  // The first and last two elements in xmlLines are not needed
+  xmlLines = xmlLines.slice(1, xmlLines.length - 2);
+
+  for (let i = 0; i < xmlLines.length; i++) {
+    // Uses regex to capture the xml attribute and element
+    // XML format: <replacement offset='OFFSET' length='LENGTH'>FIX</replacement
+    const offset: string[]|null = (/offset=\'(\d+)\'/g).exec(xmlLines[i]);
+    const length: string[]|null = (/length=\'(\d+)\'/g).exec(xmlLines[i]);
+    const fix: string[]|null =
+        (/length=\'\d+\'>(.*)<\/replacement>/g).exec(xmlLines[i]);
+
+    if (length === null || offset === null || fix === null) {
+      throw new Error('Unable to get replacement');
+    }
+    replacements[i] = {
+      offset: Number(offset[1]),
+      length: Number(length[1]),
+      fix: entities.decodeXML(fix[1])
+    };
+  }
+  return replacements;
+}
+
+/**
+ * Gets an object containing the differences between the original file and after
+ * changes have been made
+ *
+ * @param file
+ * @param replacements array of all the formatting issues within in the file
+ */
+async function getDiffObj(
+    file: string, replacements: Replacement[]): Promise<jsdiff.IUniDiff> {
+  const text = await readFilep(file, 'utf8');
+  const fixed = performFixes(text, replacements);
+  const diff =
+      jsdiff.structuredPatch(file, '', text, fixed, '', '', {context: 3});
+  jsdiff.applyPatch('diff', diff);
+  return diff;
+}
+
+/**
+ * Performs formatting fixes to the original string
+ *
+ * @param data original string
+ * @param errOffset start index of the formatting issue
+ * @param errLength length of formatting issue
+ * @param replacements string that resolves formatting issue
+ */
+function performFixes(data: string, replacements: Replacement[]) {
+  const replaced: string[] = [];
+  replaced.push(substring(data, 0, replacements[0].offset));
+  for (let i = 0; i < replacements.length - 1; i++) {
+    replaced.push(replacements[i].fix);
+    replaced.push(substring(
+        data, replacements[i].offset + replacements[i].length,
+        replacements[i + 1].offset));
+  }
+  replaced.push(replacements[replacements.length - 1].fix);
+  replaced.push(substring(
+      data,
+      replacements[replacements.length - 1].offset +
+          replacements[replacements.length - 1].length,
+      Buffer.byteLength(data, 'utf8')));
+  return replaced.join('');
+}
+
+/**
+ * Prints the lines with formatting issues
+ *
+ * @param diffs contains all information about the formatting changes
+ * @param options
+ */
+function printDiffs(diffs: jsdiff.IUniDiff, options: Options) {
```

Comment:
Suggestion: For string constructions in this file, try string interpolation.

```
const x = 2;
console.log(`${x + x} = 4`) // prints "4 = 4"

Reply:
I'm using string interpolation now. Check if it's okay